File: 2015/go/02.go
// Advent of Code 2015, Day 2
// (c) aichingert

package main

import (
	"fmt"
	"os"
	"strings"
	"strconv"
	"sort"
)


func main() {
	dat, err := os.ReadFile("../input/02")

	if err != nil {
		panic(err)
	}

	vls := strings.Split(string(dat), "\n")
	paper := 0
	ribbon := 0

	for i:=0; i < len(vls)-1; i++ {
		s := strings.Split(vls[i], "x")
		l,_ := strconv.Atoi(strings.TrimSpace(s[0]))
		w,_ := strconv.Atoi(strings.TrimSpace(s[1]))
		h,_ := strconv.Atoi(strings.TrimSpace(s[2]))
		n := []int{l,w,h}
		sort.Ints(n)

		paper += 2*n[0]*n[1] + 2*n[1]*n[2] + 2*n[2]*n[0] + n[0]*n[1]
		ribbon += 2*n[0]+2*n[1] + n[0]*n[1]*n[2]
	}

	fmt.Printf("Part 1: %d\n", paper)
	fmt.Printf("Part 1: %d\n", ribbon)
}

File: 2015/julia/02.jl
# Advent of Code 2015, day 2
# (c) aichingert

open("../input/02", "r") do f
    p1 = 0
    p2 = 0

    for l in eachline(f)
        n = sort([parse(Int64, s) for s in split(l, "x")])
        p1 += 2*n[1]*n[2] + 2*n[2]*n[3] + 2*n[1]*n[3] + n[1]*n[2]
        p2 += 2*n[1]+2*n[2] + n[1]*n[2]*n[3]
    end

    println("Part 1: " * string(p1))
    println("Part 2: " * string(p2))
end
